backend/lib/piece-indexer.js

```diff
@@ -43,12 +47,14 @@ export async function fetchDealsWithNoPayloadCid (pgPool, maxDeals) {
 async function updatePayloadInActiveDeal (pgPool, deal) {
   const updateQuery = `
     UPDATE active_deals
-    SET payload_cid = $1
-    WHERE activated_at_epoch = $2 AND miner_id = $3 AND client_id = $4 AND piece_cid = $5 AND piece_size = $6 AND term_start_epoch = $7 AND term_min = $8 AND term_max = $9 AND sector_id = $10
+    SET payload_cid = $1, payload_unretrievable = $2, last_payload_retrieval = $3
```

Comment:
We can also avoid passing down and mapping `last_payload_retrieval` value by simply using `NOW()`

```suggestion
    SET payload_cid = $1, payload_unretrievable = $2, last_payload_retrieval = NOW()
```